src/radical/saga/adaptors/cobalt/cobaltjob.py

```diff
@@ -168,46 +156,37 @@ def _cobaltscript_generator(url, logger, jd, ppn, is_cray=False, queue=None,
 
     if jd.output:
         # if working directory is set, we want stdout to end up in
-        # the working directory as well, unless it containes a specific
+        # the working directory as well, unless it contains a specific
         # path name.
-        if jd.working_directory:
-            if os.path.isabs(jd.output):
-                cobalt_params += '#COBALT --output %s\n' % jd.output
-            else:
-                # user provided a relative path for STDOUT. in this case
-                # we prepend the workind directory path before passing
-                # it on to Cobalt
-                cobalt_params += '#COBALT --output %s/%s\n' \
-                               % (jd.working_directory, jd.output)
+        if jd.working_directory and not os.path.isabs(jd.output):
+            # user provided a relative path for STDOUT, in this case
+            # we prepend the working directory path before passing
+            # it on to Cobalt
+            std_output = '%s/%s' % (jd.working_directory, jd.output)
         else:
-            cobalt_params += '#COBALT --output %s\n' % jd.output
+            std_output = jd.output
+        cobalt_params += '#COBALT --output %s\n' % std_output
 
     if jd.error:
         # if working directory is set, we want stderr to end up in
         # the working directory as well, unless it contains a specific
         # path name.
-        if jd.working_directory:
-            if os.path.isabs(jd.error):
-                cobalt_params += '#COBALT --error %s\n' % jd.error
-            else:
-                # user provided a realtive path for STDERR. in this case
-                # we prepend the workind directory path before passing
-                # it on to Cobalt
-                cobalt_params += '#COBALT --error %s/%s\n' \
-                               % (jd.working_directory, jd.error)
+        if jd.working_directory and not os.path.isabs(jd.error):
+            # user provided a relative path for STDERR, in this case
+            # we prepend the working directory path before passing
+            # it on to Cobalt
+            std_error = '%s/%s' % (jd.working_directory, jd.error)
         else:
-            cobalt_params += '#COBALT --error %s\n' % jd.error
+            std_error = jd.error
+        cobalt_params += '#COBALT --error %s\n' % std_error
 
     if jd.wall_time_limit:
         hours = int(jd.wall_time_limit / 60)
         minutes = jd.wall_time_limit % 60
         cobalt_params += '#COBALT --time %s:%s:00\n' \
             % (str(hours).zfill(2), str(minutes).zfill(2))
 
-    if queue:
-        cobalt_params += '#COBALT --queue %s\n' % queue
-    elif jd.queue:
-        cobalt_params += '#COBALT --queue %s\n' % jd.queue
+    cobalt_params += '#COBALT --queue %s\n' % (queue or jd.queue)
```

Comment:
What happens if neither is set?

Reply:
ah, valid, I misread original statement (as `if...else`), will fix, thanks